### DISCUSSION TITLE
Testing

### DISCUSSION THREAD

**Discussion by Sreeshu123** [manojnpalat/manojnpalat.github.io#11] [General]
Sir I had a query regarding testing.
We have written tests for the controllers using Mockito so far to avoid unnecessary database access and improve the test runtime. 
Is it necessary to rewrite the testcases to ensure db access for each testcase?  Or is it alright to show db connectivity using just one testcase which shows the working and the remaining testcases done by mocking db access?

**Comment by manojnpalat** [comment]
(1) no need to rewrite the testcases for each db accees (2) yes, demo for one test case which shows working with db connectivity.  The whole idea is to make sure that you get hands-on thinking about test-cases while coding.
